pkg/base/helm.go

```diff
@@ -243,57 +254,147 @@ func removeCommonPrefix(baseFiles []BaseFile) []BaseFile {
+			deps := new(HelmChartDependencies)
+			if err := yaml.Unmarshal(content, deps); err != nil {
+				return nil, errors.Wrapf(err, "failed to unmarshal %s", parentChartYaml)
+			}
+			foundSubchartInDeps := false
+			for _, dep := range deps.Dependencies {
+				if dep.Name != subchart {
+					continue
+				}
+				foundSubchartInDeps = true
+				if dep.Alias != "" {
+					aliases = append(aliases, dep.Alias)
+				} else {
+					aliases = append(aliases, dep.Name)
+				}
+			}
+			if !foundSubchartInDeps {
+				// if the subchart is not listed in the dependencies, then it is not aliased by the parent
+				aliases = append(aliases, subchart)
+			}
+		} else {
+			return nil, errors.Errorf("failed to find upstream file %s", parentChartYaml)
+		}
+
+		newBasePaths := []string{}
+		for _, basePath := range basePaths {
+			for _, alias := range aliases {
+				newBasePath := path.Join(basePath, "charts", alias)
+				newBasePaths = append(newBasePaths, newBasePath)
+			}
+		}
+		basePaths = newBasePaths
+
+		parentChartPath = path.Join(parentChartPath, "charts", subchart)
 	}
 
-	for _, upstreamFile := range u.Files {
-		basePath := strings.TrimSuffix(upstreamFile.Path, "Chart.yaml")
-		basePath = strings.TrimSuffix(basePath, "/")
-		if !basePathMap[basePath] && strings.HasSuffix(upstreamFile.Path, "Chart.yaml") {
+	return basePaths, nil
+}
+
+// Takes an input chart path and returns a list that represents the dependency tree for the chart.
+// The top-level chart is represented by an empty string.
+//  // "" => [""]
+//  // "charts/mariadb" => ["", "mariadb"]
+//  // "charts/mariadb/charts/common" => ["", "mariadb", "common"]
+func pathToCharts(path string) []string {
+	re := regexp.MustCompile(`\/?charts\/`)
+	return re.Split(path, -1)
+}
+
+// look for any sub-chart dependencies that were not rendered and add their Chart.yaml to the base files
+func helmChartBaseAppendMissingDependencies(base Base, upstreamFiles []upstreamtypes.UpstreamFile, upstreamToBasePathsMap map[string][]string, baseToUpstreamPathMap map[string]string) Base {
+	allBasePaths := getAllBasePaths("", base)
+
+	// create a map of the upstream paths that have been rendered
+	renderedUpstreamPaths := map[string]bool{}
+	for _, basePath := range allBasePaths {
+		if upstreamPath, ok := baseToUpstreamPathMap[basePath]; ok {
+			renderedUpstreamPaths[upstreamPath] = true
+		}
+	}
+
+	for _, upstreamFile := range upstreamFiles {
+		if !strings.HasSuffix(upstreamFile.Path, "Chart.yaml") {
+			continue // only care about Chart.yaml files
+		}
+		upstreamPath := strings.TrimSuffix(upstreamFile.Path, "Chart.yaml")
+		upstreamPath = strings.TrimSuffix(upstreamPath, string(os.PathSeparator))
+		if _, ok := renderedUpstreamPaths[upstreamPath]; ok {
+			continue // already rendered this upstream path
+		}
+		basePaths := upstreamToBasePathsMap[upstreamPath]
+		for _, basePath := range basePaths {
+			logger.Infof("adding missing dependency %s to base path %s\n", upstreamFile.Path, basePath)
```

Comment:
let's not log this? since it will just be noise and is not relevant to the user. or log it as debug?

Reply:
because this will always happen